### PR TITLE
Dealing with invalid xml tag ("<br>" without closing) in vml comments

### PR DESCRIPTION
**Pull request by rdoering**
This is:

```
- [x] a bugfix
- [ ] a new feature
```

Checklist:

- [x] Changes are covered by unit tests
- [x] Code style is respected
- [x] Commit message explains **why** the change is made (see https://github.com/erlang/otp/wiki/Writing-good-commit-messages)
- [x] CHANGELOG.md contains a short summary of the change
- [x] Documentation is updated as necessary

### Why this change is needed?
- Excel writes nie-lines VML-comments as "<br>"-tag without a closing "<\br>"-tag 
- This is no valid xml code
- therefore simplexml_load_string() is failing and phpShreadsheet is failing to open the file

### CODE REVIEW

**Comment by rdoering** [comment]
If the missing php 7.4 support is a problem, I could try to fix it.

**Comment by rdoering** [comment]
Any questions?

**Comment by rdoering** [comment]
Is here any progress?

**Comment by rdoering** [comment]
Travis is only failing on PHP 7.4.

**Comment by rdoering** [comment]
Any news here?

File: src/PhpSpreadsheet/Reader/Xlsx.php [hunk 1]
```diff
@@ -916,8 +900,10 @@ public function load($pFilename)
                                     $relPath = File::realpath(dirname("$dir/$fileWorksheet") . '/' . $relPath);
 
                                     try {
+                                        $secureXml = $this->securityScanner->scan($this->getFromZipArchive($zip, $relPath));
+                                        $cleanedXml = str_replace('<br>', '<br/>', $secureXml);
```
Comment (PowerKiKi) [inline src/PhpSpreadsheet/Reader/Xlsx.php]: This is way too dangerous and cannot be merged as is.

Comment (rdoering) [reply src/PhpSpreadsheet/Reader/Xlsx.php]: You're right, this solves my problem could turn an correct code like "<br></br>" into an unwanted code like "<br/></br>". Thanks for this hint I will prepare another solution.

**Comment by rdoering** [comment]
May I should change the name of the second method.

**Comment by rdoering** [comment]
Please don't hesitate to contact me, if there still is a change requested.

**Comment by rdoering** [comment]
Do you need more information?

**Comment by rdoering** [comment]
Please let me know, if I could help to push this topic forward.

**Comment by rdoering** [comment]
@PowerKiKi  Is there an issue I can help you with?

**Comment by rdoering** [comment]
I would like to support you with this issue.

**Comment by rdoering** [comment]
I am already here :-)

**Comment by rdoering** [comment]
Anything new here?

**Comment by rdoering** [comment]
Please tell me, if I can help you.

**Comment by rdoering** [comment]
@PowerKiKi 
Thanks for reviewing the code.

The difficulty was, that the function simplexml_load_string() wasn't throw an exception in default config. That is why we added the new wrapper method  simplexml_load_string().

If the method simplexml_load_string() is fails by calling the same function simplexml_load_string() coded as before, then and only then we try to parse the code as HTML-Code instead of pure XML.

`$dom = new DOMDocument;
$rc = $dom->loadHTML($string, $options);`

Unfortunately we rethrow the exception. Thats why you are right, the new code could effect the behaviour and we should remove the rethrow-statement.

> IMHO this is going too far only to allow to load an invalid file. I'd suggest fixing invalid files before loading them with PhpSpreadsheet, or maintain your fork if you encounter many of those files.

Such Excel-files with new-lines in a VML commands are written by Excel 2013 and not really invalid. A VML command contains HTML-code instead pure XML-code. This is mostly the same but not at all. (1)


(1) - https://tools.ietf.org/html/rfc7992#section-9.12

**Comment by PowerKiKi** [comment]
> then and only then we try to parse the code as HTML

That's my problem, one time we parse as XML, but sometimes we parse as HTML. This might create new issues that might be hard to resolve.

Also [Excel 2013 is EOL since 2018-10-04](https://support.microsoft.com/en-us/lifecycle/search/918) for most people. I don't think it is worth adding a risk to our codebase for something that will naturally disappear.

But again, I'll let @MarkBaker have the final word on this.

**Comment by rdoering** [comment]
Any news here?

**Comment by rdoering** [comment]
Don't hasited to ask.

**Comment by rdoering** [comment]
Any progress on this front?

**Comment by PowerKiKi** [comment]
I know this might be frustrating after all the effort you put into this, but I'll close this PR, to avoid giving hope that it might get merged.

The technical solution suggested in this PR brings too much risk and potential maintenance nightmare.

Again, if @MarkBaker has a different opinion, I'll let him re-open/merge it. But for now I'll reject it.

**Comment by rdoering** [comment]
Hi @PowerKiKi @MarkBaker 

I totally agree, it is a big step to switch form parsing comments as HTML instead of XML but, if Microsoft is writing HTML-comments, IMHO it seems to be the right approach :-).

The only risk I see is to not fail to load a file with a malformed HTML-comment, if PHP-method DOMDocument::loadHTML has is buggy. The reason for this is:

- The new approach is limited to VML-handling, it only comes to play if simplexml_load_string() is failing while loading comments.
-  We currently processed tons of file with applied changes, without any error

If you see a potential maintenance nightmare, I would like to support you with this.

Maybe it would be better create an issue-ticket first to get a better view about what how problem looks like?

**Pull request closed without merging**